File: spring-cloud-stream-demo/kafka-provider/src/main/java/com/livk/provider/SendMessage.java
package com.livk.provider;

import com.livk.stream.entity.KafkaMessage;
import lombok.RequiredArgsConstructor;
import org.springframework.cloud.stream.function.StreamBridge;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.UUID;

/**
 * <p>
 * SendMessage
 * </p>
 *
 * @author livk
 * @date 2022/2/15
 */
@Component
@RequiredArgsConstructor
public class SendMessage {

    public static final String KAFKA_TOPIC = "livk-topic";

    private final StreamBridge streamBridge;

    @Scheduled(cron = "0/5 * * * * ?")
    public void send() {
        KafkaMessage<String> message = new KafkaMessage<>();
        message.setId(UUID.randomUUID().toString());
        message.setMsg(UUID.randomUUID() + "hello");
        message.setData(UUID.randomUUID() + "|||" + UUID.randomUUID());
        streamBridge.send(KAFKA_TOPIC, message);
    }
}

File: spring-cloud-stream/kafka-provider/src/main/kotlin/com/livk/provider/SendMessage.kt
package com.livk.provider

import com.livk.stream.entity.StreamMessage
import org.springframework.cloud.stream.function.StreamBridge
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.util.UUID

/**
 * @author livk
 */
@Component
class SendMessage(private val streamBridge: StreamBridge) {
    @Scheduled(cron = "0/5 * * * * ?")
    fun send() {
        val message = StreamMessage(
            UUID.randomUUID().toString(),
            UUID.randomUUID().toString() + "hello",
            UUID.randomUUID().toString() + "|||" + UUID.randomUUID()
        )
        streamBridge.send(KAFKA_TOPIC, message)
    }

    companion object {
        const val KAFKA_TOPIC: String = "livk-topic"
    }
}
